src/Reflection/Adapter/ReflectionIntersectionType.php

```diff
@@ -7,24 +7,25 @@
 use ReflectionIntersectionType as CoreReflectionIntersectionType;
 use Roave\BetterReflection\Reflection\ReflectionIntersectionType as BetterReflectionIntersectionType;
 use Roave\BetterReflection\Reflection\ReflectionNamedType as BetterReflectionNamedType;
-use Roave\BetterReflection\Reflection\ReflectionUnionType as BetterReflectionUnionType;
 
-use function array_filter;
 use function array_map;
+use function assert;
 
 class ReflectionIntersectionType extends CoreReflectionIntersectionType
 {
     public function __construct(private BetterReflectionIntersectionType $betterReflectionType)
     {
     }
 
-    /** @return array<ReflectionNamedType> */
+    /** @return list<ReflectionNamedType> */
```

Comment:
Can we perhaps make this a `non-empty-list`? :thinking:

Reply:
All `non-empty-list` added.